pkg/kubelet/cm/helpers_linux.go

```diff
@@ -102,6 +102,18 @@ func ResourceConfigForPod(pod *v1.Pod) *ResourceConfig {
 		}
 	}
 
+	// For cpuLimits, need to compare the normal containers' resource
+	// requirement against the init-containers' and choose a bigger one.
+	// Otherwise, if a bigger init-container for a pod is provided,
+	// pod will fail to start due to "invalid argument" error when trying
+	// to write the cpu configuration into cpu.cfs_quota_us.
+	for _, container := range pod.Spec.InitContainers {
+		limits := container.Resources.Limits.Cpu().MilliValue()
+		if limits > cpuLimits {
```

Comment:
I think the init-container and regular container share the same cgoup config. As a result,  if we specify a cpu limit for init-container without specifying any cpu limit for the regular container, kubelet will ignore the init-container's cpu limit configuration. Therefore, the init-container will be unlimited as well as the regular ones. I made a change according to your suggestion, PTAL, thanks!